src/System/Posix/Thread.hsc

```diff
@@ -1,4 +1,6 @@
 -- | Bindings to the POSIX threads library.
+--
+-- Requires linking with the @-threaded@ RTS.
```

Comment:
The error isn't very good when the user forgets about this.
```
    creates 1 threads successfully
spec: schedule: re-entered unsafely.
   Perhaps a 'foreign import unsafe' should be 'safe'?
```

A runtime check in `create` would help.

Reply:
True. Will add. But I don't know of a better way to test for `-threaded` than to use `rtsSupportsBoundThreads`.